testing/regress/ecl/redislockingtest.ecl

```diff
@@ -0,0 +1,122 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+IMPORT redisServer FROM lib_redis;
+IMPORT Std;
+
+STRING server := '--SERVER=127.0.0.1:6379';
+STRING password := 'foobared';
+myRedis := redisServer(server, password);
+
+myFuncStr(STRING key) := FUNCTION
+ value := myRedis.GetString(key);
+ return value;
+END;
+myFuncUtf8(STRING key) := FUNCTION
+ value := myRedis.GetUtf8(key);
+ return value;
+END;
+myFuncUni(STRING key) := FUNCTION
+ value := myRedis.GetUnicode(key);
+ return value;
+END;
+
+getString(STRING key, STRING key2, myFuncStr func) := FUNCTION
+    value := myRedis.GetOrLockString(key);
+    RETURN IF (LENGTH(value) = 0, myRedis.SetAndPublishString(key, func(key2)), value);
+END;
+getUtf8(STRING key, STRING key2, myFuncUtf8 func) := FUNCTION
+    value := myRedis.GetOrLockUtf8(key);
+    RETURN IF (LENGTH(value) = 0, myRedis.SetAndPublishUtf8(key, func(key2)), value);
+END;
+getUnicode(STRING key, STRING key2, myFuncUni func) := FUNCTION
+    value := myRedis.GetOrLockUnicode(key);
+    RETURN IF (LENGTH(value) = 0, myRedis.SetAndPublishUnicode(key, func(key2)), value);
+END;
+
+//Test compatibiltiy between locking and normal functions.
+SEQUENTIAL(
+    myRedis.FlushDB(),
+    myRedis.SetUtf8('utf8', U'אבגדהוזחטיךכלםמןנסעףפץצקרשת'),
```

Comment:
All these tests should be using different strings.  (Did I already comment this somewhere else?)